primer/brand
# Add ComparisonTable component to Primer Brand

## Summary

- Closes https://github.com/github/primer/issues/1413
- Implementation for https://github.com/github/primer/pull/1428

Adds a `ComparisonTable` component for use in marketing pages.


🔗 [Try it out in Storybook](https://primer-2fceab1d4f-26139705.drafts.github.io/storybook/?path=/story/components-comparisontable--playground)
🔗 [See the docs](https://primer-2fceab1d4f-26139705.drafts.github.io/components/ComparisonTable)

## List of notable changes:



- **added** baseline component for `ComparisonTable`
- **fixed** ESLint configs in packages/react
- **fixed** a noisy storybook warning in dev relating to the recently deprecated `showName` property

## What should reviewers focus on?

- [Try out the examples](https://primer-2fceab1d4f-26139705.drafts.github.io/storybook/?path=/story/components-comparisontable--playground)
- 🔗 [See the docs](https://primer-2fceab1d4f-26139705.drafts.github.io/components/ComparisonTable)


## Steps to test:



1. Open [Storybook](https://primer-2fceab1d4f-26139705.drafts.github.io/storybook/?path=/story/components-comparisontable--playground) and documentation previews
1. Use the [Storybook](https://primer-2fceab1d4f-26139705.drafts.github.io/storybook/?path=/story/components-comparisontable--playground) playground controls to experiment with rendered output
1. Review the API for correctness
1. 🔗 [See the docs](https://primer-2fceab1d4f-26139705.drafts.github.io/components/ComparisonTable)


## Supporting resources (related issues, external links, etc):

- API doc https://github.com/github/primer/pull/1428

## Todo list for author

- [ ] a11y design review
- [ ] Backfill unit test coverage
- [x] Backfill visual test coverage
- [ ] Add to KitchenSink
- [ ] Add to e2e integration test fixtures

## Contributor checklist:


## Reviewer checklist:


## Screenshots:

### Default presentation in light mode
![table showing a matrix of features between Jenkins and GitHub using Storybook light mode](https://user-images.githubusercontent.com/13340707/197801764-067239c2-2e9d-45f9-b0d8-894828b5f6eb.png)

### Default presentation in dark mode
![table showing a matrix of features between Jenkins and GitHub using Storybook dark mode](https://user-images.githubusercontent.com/13340707/197801740-4249cc3d-c76f-4738-b227-90852681d506.png)


### Minimal presentation in light mode
![table showing a simplified matrix of features between Jenkins and GitHub using Storybook light mode](https://user-images.githubusercontent.com/13340707/197801722-6cab0749-9a12-438a-b486-db62de829d5c.png)


### Minimal presentation in dark mode
![table showing a simplified matrix of features between Jenkins and GitHub using Storybook dark mode](https://user-images.githubusercontent.com/13340707/197801697-1475005c-1b0e-4a59-9d57-fe47ad8612aa.png)
